textworld/generator/game.py

```diff
@@ -588,19 +608,23 @@ def __init__(self, quest: Quest, kb: KnowledgeBase) -> None:
         # Build a tree representation of the quest.
         self._tree = ActionDependencyTree(kb=self._kb,
                                           element_type=ActionDependencyTreeElement)
-        self._tree.push(quest.win_action)
+
+        for win_action in quest.win_actions:
+            self._tree.push(win_action, allow_multi_root=True)
+
         for action in quest.actions[::-1]:
             self._tree.push(action)
 
-        self._winning_policy = quest.actions + (quest.win_action,)
+        self._winning_policy = quest.actions + (quest.win_actions[0],)
```

Comment:
This might need to be re-think. It assumes the provided actions are used to make sure the first set of winning conditions are met. :-/ Maybe we also want multiple sequences of actions, i.e. one per set winning conditions? At the moment, we do allow only setting up winning conditions and provide no actions (i.e. no walkthrough).